packages/issuer/lib/functions/CredentialOffer.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  CredentialFormatEnum,
+  CredentialIssuerMetadataSupportedCredentials,
+  encodeJsonAsURI,
+  IssuerMetadata,
+  TokenErrorResponse,
+} from '@sphereon/openid4vci-common'
+import { v4 as uuidv4 } from 'uuid'
+
+export function createCredentialOfferDeeplink(preAuthorizedCode: string, issuerMetadata: IssuerMetadata): string {
```

Comment:
createCredentialOfferURI

Reply:
Where is the rename and that actual overload or optional offer object then? Right now you are just creating a state, with no control for the user to define that state. Hance why you really want to be able to also pass in the object, so you have complete control what is in there. Also you are blindly applying supported credential types from the issuer to the offer, which certainly should not happen. The URI needs to be able to only have a subset of what is available in the issuer/AS.

Reply:
changed this method to meet the criteria that we discussed